numojo/core/array_creation_routines.mojo

```diff
@@ -790,6 +790,40 @@ fn triu[dtype: DType = DType.float64](m: NDArray[dtype], k: Int = 0) raises -> N
     return result^
 
 
+fn vander[dtype: DType = DType.float64](x: NDArray[dtype], N: Optional[Int] = None, increasing: Bool = False) raises -> NDArray[dtype]:
```

Comment:
For `N`, maybe we can use the `N: Int = -1` trick 😃

`fn vander[dtype: DType = DType.float64](x: NDArray[dtype], N: Int = -1, increasing: Bool = False)`

If it is -1, then return a square array.

In this case we do not need to use `Option`.